### DISCUSSION TITLE
Develop the "auto" mode selection

### DISCUSSION THREAD

**Discussion by AlexanderRavenheart** [thingsiplay/tochd#4] [Ideas]
Since Discussions are enabled on this repository, I figured it would be a good place to launch a brainstorming session for building a refined automatic CD/DVD format selection based on a couple of factors:
- the Media Type of the game's physical release (since digital only games would not have such a field?)
  Here are examples from MobyGames:
  - CD-ROM: https://www.mobygames.com/attributes/attribute/518/
  - DVD-ROM: https://www.mobygames.com/attributes/attribute/519/
- the emulator being used to play the generated .chd files
  - performance might be degraded if using the default CD format
  - the emulator might not support the DVD format

What are your thoughts on this?

**Comment by thingsiplay** [comment]
I would like to see some reliable ways to automate the format recognition as well. These are just some random thoughts.

Such a database could be a way, but there are couple of problems in my mind: It would only work for games that are known, meaning romhacks and probably some newer translations in example would not. Also I don't know what the method of recognizing what game or system it is would be appropriate. Filenames tend to get messy and often do nor carry all needed information. But it could just default to CD in such cases, when auto could not find anything.

`7z l name.iso` would list additional information, besides the filenames (which itself could be a hint of the system) in includes. I believe this method does not work with every ISO, depending on its actual format. Also not possible with .cue/.gdi files with only bin files and such. Could get messy.
